nx/test/nx/serving_test.exs

```diff
@@ -690,27 +681,147 @@ defmodule Nx.ServingTest do
           batch = [Nx.tensor([11, 12]), Nx.tensor([13, 14])]
 
           {stream, :preprocessing!} = Nx.Serving.batched_run(config.test, batch)
-          assert Enum.to_list(stream) == [{:done, Nx.tensor([[22, 24], [26, 28]]), :metadata}]
+          assert Enum.to_list(stream) == [{:batch, Nx.tensor([[22, 24], [26, 28]]), :metadata}]
         end)
 
       t2 =
         Task.async(fn ->
           batch = [Nx.tensor([21, 22]), Nx.tensor([23, 24])]
 
           {stream, :preprocessing!} = Nx.Serving.batched_run(config.test, batch)
-          assert Enum.to_list(stream) == [{:done, Nx.tensor([[42, 44], [46, 48]]), :metadata}]
+          assert Enum.to_list(stream) == [{:batch, Nx.tensor([[42, 44], [46, 48]]), :metadata}]
         end)
 
       Task.await(t1, :infinity)
       Task.await(t2, :infinity)
 
-      assert_received {:init, :process, [[batch_keys: [:default], hooks: %{}]]}
+      assert_received {:init, :process, [[batch_keys: [:default]]]}
       assert_received {:batch, 0, batch}
       assert_received :execute
       assert batch.size == 4
       assert batch.pad == 0
     end
 
+    test "5=3+3 (oversized)", config do
+      serving =
+        Nx.Serving.new(Simple, self())
+        |> Nx.Serving.process_options(batch_size: 3)
+        |> Nx.Serving.streaming()
+
+      simple_supervised!(config, serving: serving, batch_timeout: 100)
+
+      batch =
+        Nx.Batch.stack([
+          Nx.tensor([1, 2]),
+          Nx.tensor([3, 4]),
+          Nx.tensor([5, 6]),
+          Nx.tensor([7, 8]),
+          Nx.tensor([9, 10]),
+          Nx.tensor([11, 12])
+        ])
+
+      assert Nx.Serving.batched_run(config.test, batch) |> Enum.to_list() == [
+               {:batch, Nx.tensor([[2, 4], [6, 8], [10, 12]]), :metadata},
+               {:batch, Nx.tensor([[14, 16], [18, 20], [22, 24]]), :metadata}
+             ]
```

Comment:
Should we also have a test without streaming, where we get the whole batch as output?